changelogs/client_server/newsfragments/2141.feature

```diff
@@ -0,0 +1 @@
+Add the OAuth 2.0 based authentication API, as per [MSC3861](https://github.com/matrix-org/matrix-spec-proposals/pull/3861) and its sub-proposals.
```

Comment:
My guess is that we will use this changelog for most of the PRs related to the OAuth 2.0 API.